File: basic_coding/competetive/sort/combsort.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> combsort(vector<int> array){

    int gap = array.size() * 10 / 13; //Gap factor
    while (gap > 0){
        int swapped = 0;
        for (int i = 0; i < array.size() - gap; i++){
            if (array[i + gap] < array[i]){
                swap(array[gap + i], array[i]);
                swapped = 1;
            }
        }
        if ((gap = gap * 10 / 13) == 0)
            gap = swapped;
    }
    return array;
}

int main(){

    int n;
    cin >> n;
    vector<int> A(n);
    for (int i = 0; i < n; i++)
        cin >> A[i];

    A = combsort(A);

    for (int a : A)
        cout << a << ' ';
    cout << endl;
    return 0;
}

File: competitive/sort/combsort.py
def combsort(array):
	gap = (len(array) * 10 // 13) if len(array) > 0 else 0 #gap factor = n * 10 / 13
	while gap:
		swapped = 0
		for i in range(len(array) - gap):
			if array[i + gap] < array[i]:
				array[i], array[i + gap] = array[i + gap], array[i]
				swapped = 1
		gap = (gap * 10 // 13) or swapped #(gap * 10 // 13) == 0 ? swapped : (gap * 10 // 13)
	return array
		
if __name__ == '__main__':
	a = map(int, raw_input().split())
	combsort(a)
	print a